crates/stackable-versioned-macros/src/lib.rs

```diff
@@ -65,101 +110,272 @@ mod consts;
+///     fn from(foo: v1alpha1::Foo) -> Self {
 ///         Self {
-///             bar: std::default::Default::default(),
-///             baz: __sv_foo.baz,
+///             bar: foo.prev_bar.into(), // 2
+///             baz: foo.baz,
 ///         }
 ///     }
 /// }
-/// #[automatically_derived]
+///
 /// pub mod v1beta1 {
+///     use super::*;
 ///     pub struct Foo {
-///         pub bar: usize,
+///         pub bar: usize,               // 2
 ///         pub baz: bool,
 ///     }
 /// }
-/// #[automatically_derived]
-/// #[allow(deprecated)]
-/// impl From<v1beta1::Foo> for v1::Foo {
-///     fn from(__sv_foo: v1beta1::Foo) -> Self {
+/// ```
+/// </details>
+///
+/// ### Deprecated Action
+///
+/// This action indicates that an item is deprecated in a particular version.
+/// Deprecated items are not removed.
+///
+/// ```
+/// # use stackable_versioned_macros::versioned;
+/// #[versioned(version(name = "v1alpha1"), version(name = "v1beta1"))]
+/// pub struct Foo {
+///     #[versioned(deprecated(since = "v1beta1"))]
+///     deprecated_bar: usize,
+///     baz: bool,
+/// }
+/// ```
+///
+/// <details>
+/// <summary>Generated code</summary>
+///
+/// 1. In version `v1alpha1` the field `bar` is not yet deprecated and thus uses
+///    the name without the `deprecated_` prefix.
+/// 2. In version `v1beta1` the field is deprecated and now includes the
+///    `deprecated_` prefix. It also uses the `#[deprecated]` attribute to
+///    indicate to Clippy this part of Rust code is deprecated. Therefore, the
+///    `From` implementation includes `#[allow(deprecated)]` to allow the
+///    usage of deprecated items in automatically generated code.
+///
+/// ```ignore
+/// pub mod v1alpha1 {
+///     use super::*;
+///     pub struct Foo {
+///         pub bar: usize,                     // 1
+///         pub baz: bool,
+///     }
+/// }
+///
+/// #[allow(deprecated)]                        // 2
+/// impl From<v1alpha1::Foo> for v1beta1::Foo {
+///     fn from(foo: v1alpha1::Foo) -> Self {
 ///         Self {
-///             deprecated_bar: __sv_foo.bar,
-///             baz: __sv_foo.baz,
+///             deprecated_bar: foo.bar,        // 2
+///             baz: foo.baz,
 ///         }
 ///     }
 /// }
-/// #[automatically_derived]
-/// pub mod v1 {
+///
+/// pub mod v1beta1 {
+///     use super::*;
 ///     pub struct Foo {
-///         #[deprecated = "not needed"]
+///         #[deprecated]                       // 2
 ///         pub deprecated_bar: usize,
 ///         pub baz: bool,
 ///     }
 /// }
 /// ```
+/// </details>
 ///
-/// #### Skip [`From`] generation
+/// ## Auto-generated `From` Implementations
 ///
-/// Generation of these [`From`] implementations can be skipped at the container
-/// and version level. This enables customization of the implementations if the
-/// default implementation is not sufficient.
+/// To enable smooth version upgrades of the same container, the macro
```

Comment:
Consider rewording "upgrades of the same container".

Reply:
Done in 29aed25